idx/memory/tag_query.go

```diff
@@ -26,36 +27,73 @@ var (
 // some of the following operators are non-standard and are only used
 // internally to implement certain functionalities requiring them
 
-type match uint16
+type match uint8
 
 const (
-	EQUAL      match = iota // =
-	NOT_EQUAL               // !=
-	MATCH                   // =~        regular expression
-	MATCH_TAG               // __tag=~   relies on special key __tag. non-standard, required for `/metrics/tags` requests with "filter"
-	NOT_MATCH               // !=~
-	PREFIX                  // ^=        exact prefix, not regex. non-standard, required for auto complete of tag values
-	PREFIX_TAG              // __tag^=   exact prefix with tag. non-standard, required for auto complete of tag keys
+	OP_NONE       match = iota
```

Comment:
please:
1) follow Go naming conventions. see https://stackoverflow.com/a/22688926/227168
2) do renames and cleanups separate from your logical changes. ideally as a commit preceeding the rest, or even as a separate PR (if you order your commits properly, it's always easy to make separate branches and PR's)